Blog editorial updates (#438)
OpenLiberty/blogs: posts/2020-06-03-asynchronous-programming-made-resilient-with-mpft.adoc

image::/img/blog/FT-basic-asynchronous-execution.png[Fault Tolerance asynchronous execution flow]

Sometime later, the method runs and returns a `CompletionStage` or `Future`.
Now, Fault Tolerance completes the `CompletionStage` or `Future` that was returned earlier, with the value that was returned from the method. You might be wondering, *_"What if the returned result is an exception? What can I do?"_*. Read along.

== `@Asynchronous` use cases
So `@Asynchronous` sounds great, right? But how can you use it to make your asynchronous programs resilient? Let's go through two use cases:

1. <<Applying-to-asynch-api, Applying Fault Tolerance to an API call that is already asynchronous>>
2. <<Running-methods-in-parallel, Using the `@Asynchronous` annotation to do multiple things in parallel with resilience>>

[#Applying-to-asynch-api]
=== 1. Applying Fault Tolerance to an asynchronous API call
One use of `@Asynchronous` is to apply Fault Tolerance to an asynchronous API call, which returns a `CompletionStage`. Without `@Asynchronous`, you wouldn't be able to apply Fault Tolerance to the call. Let's see why.

> In this example, we'll use the JAX-RS client API, which was introduced in JAX-RS 2.1, for calling remote REST services. We can build up a request to fetch a String from a given URL with a GET request, where the return type is a `CompletionStage` of String:

Reviewer: Technically not true.

The JAX-RS client API was introduced in JAX-RS 2.0. The reactive client APIs were added in JAX-RS 2.1.

}
----

If we call the `clientDemo()` method without any annotations, it works as we expect. We call it, receive a `CompletionStage<String>` (named `response` in the example), and then we can add an action to take when the `CompletionStage` completes:

[source,java]
----
response.thenAccept(System.out::println);
--> responseText
----

> However, if we want to use the `@Retry` annotation to retry any failed requests, the method isn't retried.

Reviewer: Not sure this sentence really makes sense. The start of the method talks about wanting to use the retry annotation, but the end of the sentence just states that the method isn't retried, as if we really had added the retry annotation, rather than just thinking about it.

How about this instead:
```suggestion
If we wanted to retry any failed requests, we might add the `@Retry` annotation to the method, but surprisingly this wouldn't work!
```
